/ekf_node:
  ros__parameters:
    # Frame IDs (already in your launch file)
    map_frame: 'map'
    odom_frame: 'odom'
    base_link_frame: 'base_link'
    
    topics:
      imu: '/imu/data'
      mag: '/imu/mag'
      gps_fix: '/gps/fix'
      gps_vel: '/gps/vel'
      odometry: '/ekf/odom'
      filtered_imu: '/ekf/filtered_imu'
      publish_acceleration: true
    
    ahrs:
      offset_samples: 100
      gain: 0.05
      gyro_range: 500.0
      accel_rejection: 10.0
      mag_rejection: 10.0
      rejection_timeout: 500
    
    kalman_filter:
      initial_pos_uncertainty: 25.0
      initial_vel_uncertainty: 1.0
      initial_bias_uncertainty: 0.01
      process_noise_pos: 0.1
      process_noise_vel: 0.01
      process_noise_bias: 0.001
      measurement_noise_pos: 25.0
      measurement_noise_vel: 0.25
    
    filters:
      lowpass_alpha_acc: 0.98
    
    earth:
      gravity: 9.8066
      radius: 6378137.0
      flattening: 0.0033528106647474805